### DISCUSSION TITLE
What about using references to files

### DISCUSSION THREAD

**Discussion by garysolebc** [CMSgov/price-transparency-guide#231] [Ideas]
![image](https://user-images.githubusercontent.com/82237927/138348143-77b4c2a7-96ae-45e5-8e65-7ef4fe8c3d00.png)

**Comment by mjbriggs** [comment]
I think that this is a fantastic idea. 

The piecemeal consumption would be much easier for people who actually want to read these files. Regarding the spirit of the law, this would be much more in line with what it wants to accomplish. Realistically, no normal person is going to read through these files to "shop prices". Rather, these files are meant to serve as a foundation for analysis and understanding of the current healthcare pricing climate. In order for this legislation to result in meaningful change, people should feel empowered to consume this plan data, despite the daunting data size. 

Reading a giant file at once may not be the easiest for someone who wants to play around with the rate files. Even if the compressed health plan file is relatively small, someone will need to decompress these files in order to make them usable. Not only could that take an inconveniently long time, it also requires that the consumer provision disk space to house these files for analysis. Presenting them in smaller chunks would allow these files to be analyzed on the fly, as the data could just be loaded into memory rather than having to be saved and then decompressed. 

I would also suggest that the reporting plans array be represented within it's own file as well.

**Reply by mjbriggs** [reply]
@shaselton-usds  Is this being considered in a meaningful way by CMS?

Leveraging references for json objects would do so much for distributing the work required to generate these files, as well as distributing the storage burden of these files.

**Comment by shaselton-usds** [comment]
@garysolebc @mjbriggs 

The updates to allow for external provider references should support the spirit of this idea. Let me know if there is any confusion with that implementation.
